<?xml version="1.0"?>
<launch>
  <!-- Median filter enabling  -->
  <arg default="true" name="use_median_filtering"/>
  <arg default="false" name="use_modified_median_filtering"/>
  <!-- Color inverse enabling  -->
  <arg default="true" name="use_color_inverse"/>
  <!-- Initial map path -->
  <arg default="$(find gazebo_heightmap_preparation)/launch/sample_images/sample_map.pgm" name="map_filepath"/>
  <!-- Prepared map and generated world saving path -->
  <arg default="$(find gazebo_heightmap_preparation)/launch/sample_images/result" name="saving_filepath"/>
  <!-- Offset to crop from left-top corner of image -->
  <arg default="1800" name="offset_x"/>
  <arg default="1800" name="offset_y"/>
  <!-- Desired width and height. They are used to compute final size of image: Gazebo hrighmap needs 2^n + 1 pixel size images, so desired size increases to nearest 2^n + 1 size. For example, 400 px turns to 513 px size.  -->
  <arg default="400" name="desired_width"/>
  <arg default="400" name="desired_height"/>
  <!-- Thresholds used to smooth out heightmap. Every pixel below "bot_trshd" turns to 0 value. Every pixel above "top_trshd" turns to 255 value -->
  <arg default="0" name="bot_trshd"/>
  <arg default="253" name="top_trshd"/>
  <!-- Filtering iterations count. The more filter iterations, the less noise level. -->
  <arg default="1" name="iter"/>

  <node args="--f $(arg map_filepath) --s $(arg saving_filepath) --offset $(arg offset_x) $(arg offset_y) --w $(arg desired_width) --h $(arg desired_height) --use_median_filtering $(arg use_median_filtering) --use_modified_median_filtering $(arg use_modified_median_filtering) --use_color_inverse $(arg use_color_inverse) --bot_trshd $(arg bot_trshd) --top_trshd $(arg top_trshd) --iter $(arg iter)" name="map_preparing_node" output="screen" pkg="gazebo_heightmap_preparation" type="map_preparing_node" required="true"/>

  <!-- Heightmap sizes in generated world file defined in meters. -->
  <arg default="60" name="world_length"/>
  <arg default="60" name="world_width"/>
  <arg default="2" name="world_height"/>

  <node args="--f $(arg saving_filepath)/prepared_map.png --l $(arg world_length) --w $(arg world_width) --h $(arg world_height)" name="world_creator" pkg="gazebo_heightmap_preparation" type="world_creator"/>
</launch>